src/app/submission/sections/sections.service.ts

```diff
@@ -0,0 +1,195 @@
+      filter((sectionObj) => hasValue(sectionObj)),
+      map((sectionObj: SubmissionSectionObject) => sectionObj.isValid),
+      distinctUntilChanged());
+  }
+
+  public isSectionActive(submissionId: string, sectionId: string): Observable<boolean> {
+    return this.submissionService.getActiveSectionId(submissionId).pipe(
+      map((activeSectionId: string) => sectionId === activeSectionId),
+      distinctUntilChanged());
+  }
+
+  public isSectionEnabled(submissionId: string, sectionId: string): Observable<boolean> {
+    return this.store.select(submissionSectionFromIdSelector(submissionId, sectionId)).pipe(
+      filter((sectionObj) => hasValue(sectionObj)),
+      map((sectionObj: SubmissionSectionObject) => sectionObj.enabled),
+      distinctUntilChanged());
+  }
+
+  public isSectionReadOnly(submissionId: string, sectionId: string, submissionScope: SubmissionScopeType): Observable<boolean> {
+    return this.store.select(submissionSectionFromIdSelector(submissionId, sectionId)).pipe(
+      filter((sectionObj) => hasValue(sectionObj)),
+      map((sectionObj: SubmissionSectionObject) => {
+        return isNotEmpty(sectionObj.visibility)
+          && sectionObj.visibility.other === 'READONLY'
+          && submissionScope !== SubmissionScopeType.WorkspaceItem
+      }),
+      distinctUntilChanged());
+  }
+
+  public isSectionAvailable(submissionId: string, sectionId: string): Observable<boolean> {
+    return this.store.select(submissionObjectFromIdSelector(submissionId)).pipe(
+      filter((submissionState: SubmissionObjectEntry) => isNotUndefined(submissionState)),
+      map((submissionState: SubmissionObjectEntry) => {
+        return isNotUndefined(submissionState.sections) && isNotUndefined(submissionState.sections[sectionId]);
+      }),
+      distinctUntilChanged());
+  }
+
+  public addSection(submissionId: string,
+                    sectionId: string) {
+    this.store.dispatch(new EnableSectionAction(submissionId, sectionId));
+    const config: ScrollToConfigOptions = {
+      target: sectionId,
+      offset: -70
+    };
+
+    this.scrollToService.scrollTo(config);
+  }
+
+  public removeSection(submissionId: string, sectionId: string) {
+    this.store.dispatch(new DisableSectionAction(submissionId, sectionId))
+  }
+
+  public updateSectionData(submissionId: string, sectionId: string, data, errors = []) {
+    if (isNotEmpty(data)) {
+      const isAvailable$ = this.isSectionAvailable(submissionId, sectionId);
+      const isEnabled$ = this.isSectionEnabled(submissionId, sectionId);
+
+      combineLatest(isAvailable$, isEnabled$).pipe(
+        take(1),
+        filter(([available, enabled]: [boolean, boolean]) => available))
+        .subscribe(([available, enabled]: [boolean, boolean]) => {
+          if (!enabled) {
+            this.translate.get('submission.sections.general.metadata-extracted-new-section', {sectionId})
+              .pipe(take(1))
+              .subscribe((m) => {
```

Comment:
Nested subscribes can likely be rewritten using switchMap.

Reply:
please take a look to this stackoverflow answer https://stackoverflow.com/questions/40861494/angular2-unsubscribe-from-http-observable-in-service
there is no need to unsubscribe in a service

Reply:
You're right about unsubscribing in services. However, I commented on the fact that there was a subscribe inside another subscribe. This is an anti-pattern, and it can be rewritten using switchMap, mergeMap or flatMap. More info here:
- https://blog.strongbrew.io/rxjs-best-practices-in-angular/#avoiding-nested-subscribes
- https://medium.com/@paynoattn/3-common-mistakes-i-see-people-use-in-rx-and-the-observable-pattern-ba55fee3d031#ad2b
- http://www.syntaxsuccess.com/viewarticle/fixing-rxjs-anti-patterns

Reply:
removed nested subscription